### PR TITLE
AG-12655 Skip unordered category animations

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-12655

### CODE REVIEW

File: packages/ag-charts-community/src/chart/axis/categoryAxis.ts [hunk 1]
```diff
@@ -30,6 +30,52 @@ export class CategoryAxis<
...
+        return true;
+    }
+
+    private categoryAnimatable = true;
+    protected override calculateDomain() {
+        let normalizedDomain: any[] = [];
+
+        let categoryAnimatable = true;
+        for (const series of this.boundSeries) {
+            if (!this.includeInvisibleDomains && !series.isEnabled()) continue;
```
Comment (iMoses) [inline packages/ag-charts-community/src/chart/axis/categoryAxis.ts:58]: nit: `includeInvisibleDomains` is always true in category axes, so this always returns false

File: packages/ag-charts-community/src/chart/axis/categoryAxis.ts [hunk 2]
```diff
@@ -30,6 +30,52 @@ export class CategoryAxis<
...
 
+    private domainOrderedToNormalizedDomain(seriesDomain: any[], normalizedDomain: any[]) {
+        let normalizedIndex = -1;
+        for (let seriesIndex = 0; seriesIndex < seriesDomain.length; seriesIndex += 1) {
+            const value = seriesDomain[seriesIndex];
+            const normalizedNextIndex = normalizedDomain.indexOf(value);
+
+            if (normalizedNextIndex === -1) {
+                // All subsequent values must be extending (i.e. appending to) the normalized domain
+                normalizedIndex = Number.MAX_SAFE_INTEGER;
```
Comment (iMoses) [inline packages/ag-charts-community/src/chart/axis/categoryAxis.ts]: nit: use `Infinity` to keep consistent with similar logic in our codebase

**Comment by iMoses** [comment]
Can we extract this code into `cartesianChart.ts`?
It'll help us by reducing coupling between the axis and series classes, and will keep logic closer to where it's supposed to be.

File: packages/ag-charts-community/src/chart/axis/categoryAxis.ts (hunk 2 above)
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/axis/categoryAxis.ts]:
```suggestion
                normalizedIndex = Infinity;
```

File: packages/ag-charts-community/src/chart/axis/categoryAxis.ts (hunk 1 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/axis/categoryAxis.ts:58]: It's to match the original logic - in case this value ever changes

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**